src/player.cpp

```diff
@@ -1898,21 +1898,27 @@ player::wear( item_location item_wear, bool interactive )
     return result;
 }
 
+int player::get_lift_str() const
+{
+    int str = get_str();
+    if( has_trait( trait_id( "STRONGBACK" ) ) ) {
+        str *= 1.35;
+    } else if( has_trait( trait_id( "BADBACK" ) ) ) {
+        str /= 1.35;
+    }
+    return str;
+}
+
 template <typename T>
 bool player::can_lift( const T &obj ) const
 {
     // avoid comparing by weight as different objects use differing scales (grams vs kilograms etc)
-    int str = get_str();
+    int str = get_lift_str();
     if( mounted_creature ) {
         const auto mons = mounted_creature.get();
         str = mons->mech_str_addition() == 0 ? str : mons->mech_str_addition();
```

Comment:
`str` could be altered here in the original code. So lift adjusted strength is not necessarily always based on `get_str()`. `player::get_lift_str()` should be changed to something like `player::adjust_life_str(int str)` that takes an integer `str` and calculates adjusted strength based on the integer parameter given.

Reply:
uploaded